src/test/groovy/graphql/execution/instrumentation/fieldvalidation/FieldValidationTest.groovy

```diff
@@ -0,0 +1,313 @@
+
+                // parent check
+                assert linkLink.getParentFieldAndArguments().getPath().toString() == "/field1/informationLink/informationLink"
+                assert linkLink.getParentFieldAndArguments().getParentFieldAndArguments().getPath().toString() == "/field1/informationLink"
+                assert linkLink.getParentFieldAndArguments().getParentFieldAndArguments().getParentFieldAndArguments().getPath().toString() == "/field1"
+
+
+                return [
+                        validationEnvironment.mkError("Made up some error here", field1Args)
+                ]
+            }
+        }
+
+        when:
+
+        setupExecution(validator, query, variables)
+
+        then:
+        def abortExecutionException = thrown(AbortExecutionException)
+        def errors = abortExecutionException.getUnderlyingErrors()
+        errors.size() == 1
+        errors[0].getMessage() == "Made up some error here"
+        errors[0].getLocations().size() == 1
+        errors[0].getPath() == ["field1"]
+    }
+
+    def "test SimpleFieldAndArgumentsValidator"() {
+        given:
+
+        def variables = [
+                stringVar   : "stringValue",
+                intVar      : 666,
+                inputDataVar: [string: "string", int: 0],
+                idVar       : "ID12345",
+                fmt1Var     : "fmt1Value"
+        ]
+
+        SimpleFieldValidation validation = new SimpleFieldValidation()
+                .addRule(ExecutionPath.parse("/field1"),
+                { fieldAndArguments, env -> err("Not happy Jan!", env, fieldAndArguments) })
+                .addRule(ExecutionPath.parse("/field1/informationLink/informationLink/informationString"),
+                { fieldAndArguments, env -> err("Also not happy Jan!", env, fieldAndArguments) })
+                .addRule(ExecutionPath.parse("/does/not/exist"),
+                { fieldAndArguments, env -> err("Wont happen", env, fieldAndArguments) })
+
+
+        when:
+        setupExecution(validation, query, variables)
+
+        then:
+        def abortExecutionException = thrown(AbortExecutionException)
+        def errors = abortExecutionException.getUnderlyingErrors()
+
+        errors.size() == 2
+        errors[0].getMessage() == "Not happy Jan!"
+        errors[0].getLocations().size() == 1
+        errors[0].getPath() == ["field1"]
+
+        errors[1].getMessage() == "Also not happy Jan!"
+    }
+
+    def "test fragment value uniqueness support is handled"() {
+        given:
+
+        def variables = [
+                stringVar   : "stringValue",
+                intVar      : 666,
+                inputDataVar: [string: "string", int: 0],
+                idVar       : "ID12345",
+                fmt1Var     : "ok"
+        ]
+
+        def query = '''
+        query VarTest($stringVar : String, $intVar : Int, $inputDataVar : InputData, $idVar : ID, $fmt1Var : String) {
```

Comment:
same comments as before: I don't think this is a valid query example, because it will fail general validation.